crates/net/eth-wire/src/handshake.rs

```diff
@@ -135,39 +135,41 @@ where
 
         // Validate peer response
         match msg.message {
-            EthMessage::Status(their_status) => {
+            EthMessage::Status(their_status_message) => {
                 trace!("Validating incoming ETH status from peer");
-
-                if status.genesis != their_status.genesis {
+                let their_genesis = their_status_message.genesis();
```

Comment:
pedantic style nit: we can avoid the extra tmp values and instead rely entirely on functions, these are const anyway and it's fine to call them again for the error message